plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/version/KubernetesVersionManagerImpl.java

```diff
@@ -0,0 +1,377 @@
+            if (templatesZoneVOs.isEmpty()) {
+                DataCenterVO zone = dataCenterDao.findById(zoneId);
+                throw new InvalidParameterValueException(String.format("ISO ID: %s is not available for zone ID: %s", template.getUuid(), zone.getUuid()));
+            }
+        }
+    }
+
+    private void deleteKubernetesVersionIso(long templateId) throws IllegalAccessException, NoSuchFieldException,
+            IllegalArgumentException {
+        DeleteIsoCmd deleteIsoCmd = new DeleteIsoCmd();
+        deleteIsoCmd = ComponentContext.inject(deleteIsoCmd);
+        Field f = deleteIsoCmd.getClass().getDeclaredField("id");
+        f.setAccessible(true);
+        f.set(deleteIsoCmd, templateId);
+        templateService.deleteIso(deleteIsoCmd);
+    }
+
+    @Override
+    public ListResponse<KubernetesSupportedVersionResponse> listKubernetesSupportedVersions(final ListKubernetesSupportedVersionsCmd cmd) {
+        if (!KubernetesClusterService.KubernetesServiceEnabled.value()) {
+            throw new CloudRuntimeException("Kubernetes Service plugin is disabled");
+        }
+        final Long versionId = cmd.getId();
+        final Long zoneId = cmd.getZoneId();
+        String minimumSemanticVersion = cmd.getMinimumSemanticVersion();
+        final Long minimumKubernetesVersionId = cmd.getMinimumKubernetesVersionId();
+        if (!Strings.isNullOrEmpty(minimumSemanticVersion) && minimumKubernetesVersionId != null) {
+            throw new CloudRuntimeException(String.format("Both parameters %s and %s can not be passed together", ApiConstants.MIN_SEMANTIC_VERSION, ApiConstants.MIN_KUBERNETES_VERSION_ID));
+        }
+        if (minimumKubernetesVersionId != null) {
+            KubernetesSupportedVersionVO minVersion = kubernetesSupportedVersionDao.findById(minimumKubernetesVersionId);
+            if (minVersion == null) {
+                throw new InvalidParameterValueException(String.format("Invalid %s passed", ApiConstants.MIN_KUBERNETES_VERSION_ID));
+            }
+            minimumSemanticVersion = minVersion.getSemanticVersion();
+        }
+        List <KubernetesSupportedVersionVO> versions = new ArrayList<>();
+        if (versionId != null) {
+            KubernetesSupportedVersionVO version = kubernetesSupportedVersionDao.findById(versionId);
+            if (version != null && (zoneId == null || version.getZoneId() == null || version.getZoneId().equals(zoneId))) {
+                versions.add(version);
+            }
+        } else {
+            if (zoneId == null) {
+                versions = kubernetesSupportedVersionDao.listAll();
+            } else {
+                versions = kubernetesSupportedVersionDao.listAllInZone(zoneId);
+            }
+        }
+        // Filter versions for minimum Kubernetes version
+        versions = filterKubernetesSupportedVersions(versions, minimumSemanticVersion);
```

Comment:
this comment doesn't add to the statement below

Reply:
removed comemnt